### ISSUE TITLE
【Version 3.0.0】 File name suffix disappears

### ISSUE THREAD

**Issue by macera** [carrierwaveuploader/carrierwave#2672]
File name suffix disappears from v3.0.0.rc.

```
class SampleUser < ApplicationRecord
  mount_uploader :filename, SampleAvatarUploader
end

class SampleAvatarUploader < CarrierWave::Uploader::Base
  storage :file
  def store_dir
    "uploads/#{model.class.to_s.underscore}/#{mounted_as}/#{model.id}"
  end
  def default_url(*args)
    "/images/fallback/" + [version_name, "default.png"].compact.join('_')
  end
  def extension_allowlist
    %w(jpg jpeg gif png)
  end
end
```
```
require 'rails_helper'

RSpec.describe SampleAvatarUploader do

  describe '#upload' do
    let(:tmp_sample_file_dir) { 'tmp/avatars/' }
    let(:sample1_file_path) { Rails.root.join(tmp_sample_file_dir, 'sample.jpg') }
    let(:sample3_file_path) { Rails.root.join(tmp_sample_file_dir, 'sample(3).jpg') }
    let(:sample4_file_path) { Rails.root.join(tmp_sample_file_dir, 'sample(4).jpg') }
    it '' do
      avatars =  []
      avatars << SampleUser.create(filename: Rack::Test::UploadedFile.new(sample1_file_path))
      avatars << SampleUser.create(filename: Rack::Test::UploadedFile.new(sample3_file_path))
      avatars << SampleUser.create(filename: Rack::Test::UploadedFile.new(sample4_file_path))

      file_names =  avatars.map{ |a| File.basename(a.filename.to_s) }
      expect(file_names).to contain_exactly('sample.jpg', 'sample(3).jpg', 'sample(4).jpg')
    end
  end
end
```

```
Failures:

  1) SampleAvatarUploader#upload is expected to contain exactly "sample.jpg", "sample(3).jpg", and "sample(4).jpg"
     Failure/Error: expect(file_names).to contain_exactly('sample.jpg', 'sample(3).jpg', 'sample(4).jpg')
     
       expected collection contained:  ["sample(3).jpg", "sample(4).jpg", "sample.jpg"]
       actual collection contained:    ["sample.jpg", "sample.jpg", "sample.jpg"]
       the missing elements were:      ["sample(3).jpg", "sample(4).jpg"]
       the extra elements were:        ["sample.jpg", "sample.jpg"]
     # ./spec/uploaders/sample_avatar_uploader_spec.rb:17:in `block (3 levels) in <top (required)>'
```

v2.2.4
```
SampleAvatarUploader
  #upload
    is expected to contain exactly "sample.jpg", "sample(3).jpg", and "sample(4).jpg"

Finished in 0.11796 seconds (files took 10.66 seconds to load)
1 example, 0 failures
```

probably related to this issue #2670

**Comment by mshibuya** [comment]
Sorry for the issue, this is serious. I'll look into this very soon.

**Comment by mshibuya** [comment]
My initial impression was this issue was causing stored files to be overwritten, but actually it was not. So it's not that serious but still it is inconvenient.
I made a fix as d68a1111cfae4309d703caa19d9c19226bc01686 and it is already in the master, so please try.
